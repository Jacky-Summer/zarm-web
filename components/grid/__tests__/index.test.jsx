import React from 'react';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Row, Col } from '../index';

describe('Gird', () => {
  it('renders Row,Col correctly', () => {
    const wrapper = render(
      <Row>
        <Col>col</Col>
      </Row>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Row when typeof gutter is number', () => {
    const wrapper = render(
      <Row gutter={8}>
        <Col span={12}>col-12</Col>
        <Col span={12}>col-12</Col>
      </Row>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Row when typeof gutter is array', () => {
    const wrapper = render(
      <Row gutter={[8, 10]}>
        <Col span={12}>col-12</Col>
        <Col span={12}>col-12</Col>
      </Row>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('check Col flex attribute', () => {
    const wrapper = render(
      <>
        <Row gutter={[0, 10]}>
          <Col flex={2}>
            <div>2 / 5</div>
          </Col>
          <Col flex={3}>
            <div>3 / 5</div>
          </Col>
        </Row>
        <Row gutter={[0, 10]}>
          <Col flex="100px">
            <div>100px</div>
          </Col>
          <Col flex="auto">
            <div>Fill Rest</div>
          </Col>
        </Row>
        <Row gutter={[0, 10]}>
          <Col flex="1 1 200px">
            <div>1 1 200px</div>
          </Col>
          <Col flex="0 1 300px">
            <div>0 1 300px</div>
          </Col>
        </Row>
      </>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
