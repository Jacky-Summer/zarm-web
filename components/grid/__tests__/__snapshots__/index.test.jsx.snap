// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gird check Col flex attribute 1`] = `
Array [
  <div
    class="zw-row"
    style="margin-top:-5px;margin-bottom:5px"
  >
    <div
      class="zw-col"
      style="padding-top:5px;padding-bottom:5px;flex:2 2 auto"
    >
      <div>
        2 / 5
      </div>
    </div>
    <div
      class="zw-col"
      style="padding-top:5px;padding-bottom:5px;flex:3 3 auto"
    >
      <div>
        3 / 5
      </div>
    </div>
  </div>,
  <div
    class="zw-row"
    style="margin-top:-5px;margin-bottom:5px"
  >
    <div
      class="zw-col"
      style="padding-top:5px;padding-bottom:5px;flex:0 0 100px"
    >
      <div>
        100px
      </div>
    </div>
    <div
      class="zw-col"
      style="padding-top:5px;padding-bottom:5px;flex:auto"
    >
      <div>
        Fill Rest
      </div>
    </div>
  </div>,
  <div
    class="zw-row"
    style="margin-top:-5px;margin-bottom:5px"
  >
    <div
      class="zw-col"
      style="padding-top:5px;padding-bottom:5px;flex:1 1 200px"
    >
      <div>
        1 1 200px
      </div>
    </div>
    <div
      class="zw-col"
      style="padding-top:5px;padding-bottom:5px;flex:0 1 300px"
    >
      <div>
        0 1 300px
      </div>
    </div>
  </div>,
]
`;

exports[`Gird renders Row when typeof gutter is array 1`] = `
<div
  class="zw-row"
  style="margin-left:-4px;margin-right:-4px;margin-top:-5px;margin-bottom:5px"
>
  <div
    class="zw-col zw-col-span-12"
    style="padding-left:4px;padding-right:4px;padding-top:5px;padding-bottom:5px"
  >
    col-12
  </div>
  <div
    class="zw-col zw-col-span-12"
    style="padding-left:4px;padding-right:4px;padding-top:5px;padding-bottom:5px"
  >
    col-12
  </div>
</div>
`;

exports[`Gird renders Row when typeof gutter is number 1`] = `
<div
  class="zw-row"
  style="margin-left:-4px;margin-right:-4px"
>
  <div
    class="zw-col zw-col-span-12"
    style="padding-left:4px;padding-right:4px"
  >
    col-12
  </div>
  <div
    class="zw-col zw-col-span-12"
    style="padding-left:4px;padding-right:4px"
  >
    col-12
  </div>
</div>
`;

exports[`Gird renders Row,Col correctly 1`] = `
<div
  class="zw-row"
>
  <div
    class="zw-col"
  >
    col
  </div>
</div>
`;
